test/unit/core/adapters/personalization.ts

```diff
@@ -0,0 +1,380 @@
+      const state: any = {
+        allIds: [{
+          variant: 'color',
+          key: 'blue',
+        }],
+        byId: {
+          color: {
+            blue: {
+              lastUsed: 11111
+            }
+          }
+        }
+      };
+      const key = 'test';
+
+      const result = Adapter.transformToBrowser(state, key);
+
+      expect(result).to.deep.equal({
+        allIds: [{
+          variant: 'color',
+          key: 'blue',
+          lastUsed: 11111
+        }]
+      });
+    });
+  });
+
+  describe('transformFromBrowser()', () => {
+    it('should filter allIds that are older than the set time', () => {
+      const oneDayInSec = 86400;
+      const now = Date.now() / 1000;
+      const expiry = 30;
+      const browserStorage = {
+        allIds: [{ variant: 'color', key: 'blue', lastUsed: now - oneDayInSec * 5 },
+                 { variant: 'brand', key: 'Shoe', lastUsed: now - oneDayInSec * 6 },
+                 { variant: 'color', key: 'red', lastUsed: now - oneDayInSec * 21 },
+                 { variant: 'brand', key: 'Nike', lastUsed: now - oneDayInSec * 33 }]
+      };
+      const biasFromBrowser = {
+        allIds: [{ variant: 'color', key: 'blue' }, { variant: 'brand', key: 'Shoe'},
+                 { variant: 'color', key: 'red' }],
+        byId: {
+          brand: {
+            Shoe: {
+              lastUsed: now - oneDayInSec * 6
+            }
+          },
+          color: {
+            blue: {
+              lastUsed: now - oneDayInSec * 5
+            },
+            red: {
+              lastUsed: now - oneDayInSec * 21
+            }
+          }
+        }
+      };
+      const realTimeBiasing = 'real';
+      const conf = { personalization: { realTimeBiasing } };
+      const state: any = 's';
+      const config = stub(Selectors, 'config').returns(conf);
+      const extractExpiry = stub(ConfigAdapter, 'extractRealTimeBiasingExpiry').returns(expiry);
+      const pruneBiases = stub(Adapter, 'pruneBiases').returnsArg(0);
+
+      const result = Adapter.transformFromBrowser(browserStorage, state);
+
+      expect(result).to.deep.equal(biasFromBrowser);
+      expect(config).to.be.calledWithExactly(state);
+      expect(extractExpiry).to.be.calledWithExactly(conf);
+      expect(pruneBiases).to.be.calledTwice.and
+        .calledWithExactly(biasFromBrowser.allIds, 'color', 2, realTimeBiasing).and
+        .calledWithExactly(biasFromBrowser.allIds, 'brand', 1, realTimeBiasing);
+    });
+  });
+
+  describe('pruneBiases()', () => {
+    const arr = [
+      { variant: 'a', key: 1},
+      { variant: 'b', key: 2},
+      { variant: 'b', key: 5},
+      { variant: 'c', key: 3},
+    ];
+
+    it('shoud remove last element of a specific variant if it exists ', () => {
+      const newArr = [
+        { variant: 'a', key: 1 },
+        { variant: 'b', key: 2 },
+        { variant: 'c', key: 3 },
+      ];
+
+      const ret = Adapter.pruneBiases(<any>arr, 'b', 2, <any>{ attributes: {
```

Comment:
rather than cast to `<any>` here, just set the type of `arr` to `any[]` at declaration